src/decimal/mod.rs

```diff
@@ -285,17 +285,14 @@ macro_rules! dec_impl {
     (impl_trait_from_float; $($t:ty),*) => {$(
         impl<T, P> From<$t> for GenericDecimal<T, P>
         where
-            T: Clone + GenericInteger,
+            T: Copy + Clone + GenericInteger + FromPrimitive,
```

Comment:
The `Copy` requirement is not required, but helpful. The `FromPrimitive` is required though (hence maybe the version change to `0.8` to avoid breaking uses without the `FromPrimitive` constraint).

Reply:
The `Copy` trait would prevent us from using `BigUint` here (which is on heap).
Regarding `FromPrimitive` - yes, if we cannot make it without `FromPrimitive`, then we'll have to make `0.8`.

Reply:
I don't see how we can avoid the `FromPrimitive` constraint because we need to convert the computed new value into the type requested: `let numer: T = match T::from_f64(new_val.into()) {...}`.